src/realm/object-store/c_api/error.cpp

```diff
@@ -297,3 +311,13 @@ RLM_EXPORT bool realm_wrap_exceptions(void (*func)()) noexcept
         return true;
     });
 }
+
+RLM_API void realm_register_user_code_callback_error(void* usercode_error) noexcept
+{
+    realm::c_api::wrap_err([=]() {
+        // register a callback failed exception storing the ptr to the user code error passed by the SDK
+        // when this excpetion is caught the original error can be processed
+        throw realm::c_api::CallbackFailed(usercode_error);
+        return false;
```

Comment:
Does this really need to be returning false if we are throwing in the line before it?